addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -3371,6 +3379,13 @@ const SnippetOptionWidget = Widget.extend({
 
         if (params.extraClass) {
             this.$target.toggleClass(params.extraClass, hasUserValue);
+            if (hasUserValue) {
+                // Might have changed because of the class.
+                for (const cssProp of cssProps) {
+                    this.$target[0].style.removeProperty(cssProp);
+                }
```

Comment:
Actually, why do you need this? If the values don't align with the styles, they will be overridden, and if they do, `applyCSS()` will return `false`. (cf. the call to `weUtils.areCssValuesEqual()`)

Reply:
Yes, but not for multiple values: if at least one of the values is not the default one, one of the `applyCSS()` returns `true` - and in that case the `extraClass` is added which might impact the values that did return `false`, and which wouldn't with the `extraClass`.

Reply:
Yes, but isn't your call to `applyAllCSS()` on the next line enough? It's the fact that you remove all the properties that I don't get.

Reply:
Here is how I see the two scenarios.
Without reset:
```md
- set x: 1
- add class extra: {x: 1}
// do not reset x
- `applyCSS` has no effect because `areCssValuesEqual`
=> x remains set in the inline style - while it is useless thanks to the class
```
With reset:
```md
- set x: 1
- add class extra: {x: 1}
- reset x
- `applyCSS` has no effect because `areCssValuesEqual`
=> x is not set in the inline style, but it is ok because it is applied by the extra class
```